framework/include/actions/ActionWarehouse.h

```diff
@@ -206,7 +214,9 @@ class ActionWarehouse :
   /// The Factory that builds Actions
   ActionFactory & _action_factory;
   /// Pointers to the actual parsed input file blocks
-  std::map<std::string, std::vector<Action *> > _action_blocks;
+  std::map<std::string, std::list<Action *> > _action_blocks;
+  /// Action blocks that have been requested
+  std::map<std::string, std::vector<Action *> > _requested_action_blocks;
```

Comment:
This would actually be a textbook case for mutable, but if the member function constness is not required it doesn't matter much